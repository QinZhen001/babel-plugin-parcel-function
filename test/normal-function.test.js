const {testPlugin, removeSpace} = require("./util/util")

const code1 = `
async function f() {
    await aaa();
    await bbb();
    console.log('eee');
    return 'ddd';
  }
`


const code2 = `
const test = async function(aaa,bbb) {
    await aaa();
    await bbb();
  }
`

// --------------------------------------

describe("普通函数", () => {
  test("async函数", () => {
    const res = testPlugin(code1, [], [], {
      prefixName: ["aaa", "bbb", "ccc", "ddd"]
    })
    expect(res).toMatchSnapshot();
  })
  test("赋值", () => {
    const res = testPlugin(code2, [], [], {})
    expect(res).toMatchSnapshot();
  })
})


