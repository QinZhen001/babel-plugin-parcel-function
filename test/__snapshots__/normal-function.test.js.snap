// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`普通函数 async函数 1`] = `
"
async function f() {
    if (aaa.bbb.ccc.ddd.f) {
        aaa.bbb.ccc.ddd.f.call(this);
    } else {
        await aaa();
        await bbb();
        console.log('eee');
        return 'ddd';
    }
}"
`;

exports[`普通函数 赋值 1`] = `
"
const test = async function (aaa, bbb) {
    if (global.hotUpdate) {
        global.hotUpdate.call(this, aaa, bbb);
    } else {
        await aaa();
        await bbb();
    }
};"
`;
